pyartcd/pyartcd/pipelines/build_sync.py

```diff
@@ -395,13 +407,18 @@ async def build_sync(runtime: Runtime, version: str, assembly: str, publish: boo
     try:
         await pipeline.run()
 
-    except (RuntimeError, ChildProcessError):
+    except (RuntimeError, ChildProcessError, ViableFalseError) as e:
         # Only for 'stream' assembly, track failure to enable future notifications
         if assembly == 'stream':
             await pipeline.handle_failure()
 
-        # Re-reise the exception to make the job as failed
-        raise
+        # check if error is viable false, if it is set exit code to 2 to indicate partial success
+        if isinstance(e, ViableFalseError):
+            runtime.logger.error(f'Exiting with code 2 (partial success): {e}')
+            sys.exit(2)
```

Comment:
Optional: maybe we can use some standard Unix exit code. Took a quick look [here](https://tldp.org/LDP/abs/html/exitcodes.html) but haven't found a good match for our use case at first glance.